src/Services/LightningService.cs

```diff
@@ -194,12 +184,19 @@ public async Task OpenChannel(ChannelOperationRequest channelOperationRequest)
 
             var combinedPSBT = GetCombinedPsbt(channelOperationRequest, _logger);
 
+
             //32 bytes of secure randomness for the pending channel id (lnd)
             var pendingChannelId = RandomNumberGenerator.GetBytes(32);
             var pendingChannelIdHex = Convert.ToHexString(pendingChannelId);
 
             try
             {
+                var virtualSize=combinedPSBT.GetGlobalTransaction().GetVirtualSize()+22; //22 bytes for the 1 segwit output
+                var feeRateResult = await LightningHelper.GetFeeRateResult(network, _nbXplorerService);
+
+                var totalFees = new Money(virtualSize * feeRateResult.FeeRate.SatoshiPerByte, MoneyUnit.Satoshi);
+
+                long fundingAmount = channelOperationRequest.Changeless ? channelOperationRequest.SatsAmount - totalFees : channelOperationRequest.SatsAmount;
```

Comment:
```suggestion
                var fundingAmount = channelOperationRequest.Changeless ? channelOperationRequest.SatsAmount - totalFees : channelOperationRequest.SatsAmount;
```

Reply:
Can't do this because of an ambiguous cast. All of this can be casted to long or to Money